src/parsing/parser.h

```diff
@@ -128,14 +133,20 @@ struct ParserTypes<Parser> {
   using AstRawString = AstRawString;
 };
 
-class V8_EXPORT_PRIVATE Parser : public NON_EXPORTED_BASE(ParserBase<Parser>) {
+template <typename Impl>
+class V8_EXPORT_PRIVATE AbstractParser
```

Comment:
best name i could come up with for now. open to suggestions.

Reply:
I feel like it's a fine name given that nobody is instantiating it directly.